src/compat.rs

```diff
@@ -28,10 +39,10 @@ where
     T::Error: fmt::Debug,
 {
     fn set_low(&mut self) -> Result<(), Self::Error> {
-        self.0.set_low()
+        self.0.set_low().map_err(CompatError)
     }
 
     fn set_high(&mut self) -> Result<(), Self::Error> {
-        self.0.set_high()
+        self.0.set_low().map_err(CompatError)
```

Comment:
There's a copy-paste error here:
```suggestion
        self.0.set_high().map_err(CompatError)
```